smarts/core/sumo_road_network.py

```diff
@@ -143,13 +138,9 @@ def _shift_coordinates(cls, net_file_path, shifted_path):
         return False
 
     @classmethod
-    def from_file(
-        cls,
-        net_file,
-        shift_to_origin=False,
-        default_lane_width=None,
-        lanepoint_spacing=None,
-    ):
+    def from_spec(cls, map_spec: MapSpec, shift_to_origin: bool = False):
+        net_file = map_spec.source
```

Comment:
Can we safely assume here that `map_spec.source` can be a path to the network file now as opposed to a path to scenario directory? I thought you told me that a user can set `MapSpec` when creating the Scenario?

Reply:
Yeah, I suppose we should add `map.net.xml` to it here (notwithstanding what's done in `default_map_builder.py`) if it's still just a folder, just in case we arrive here having not gone through the default builder.  Good point!  Will do.

Reply:
Done.